docs/reference/rest-api/common-parms.asciidoc

```diff
@@ -1309,6 +1309,6 @@ tag::rrf-window-size[]
 This value determines the size of the individual result sets per
 query. A higher value will improve result relevance at the cost of performance. The final
 ranked result set is pruned down to the search request's <<search-size-param, size>>.
-`window_size` must be greater than or equal to `size` and greater than or equal to `1`.
+`window_size` must be greater than or equal to `from + size` and greater than or equal to `1`.
```

Comment:
Discussing Ben's comment and taking another look at this, I'm not sure if we need to update docs here (and the check in `SearchRequest`). If we are to operate on `(from + window_size)` then `window_size` would just be an override for `size`, i.e. _fetch the window\_size docs **after** already consuming **from** docs_.

So in order to support pagination only through the `from` param, `window_size` should probably just be compared against size. (i.e. one shouldn't have to change both `from` and `window_size` to do pagination). 

Will see to update this accordingly.

Reply:
On the contrary though, for LTR we mention that: 

> To prevent this, the `window_size` parameter is mandatory for LTR rescorers and should be greater than or equal to `from + size`.

Reply:
While this isn't probably such an issue for `rrf` I wonder if we should have a more "standardized"  and generic approach here with other potential rankers.

Reply:
The key for rescorers in general is that the `window_size` doesn't change as you paginate. We don't enforce anything directly in rescorers and I don't think we should.

Similarly, I don't think we should enforce anything with retrievers. We can document the behavior & situations where RRF might paginate things strangely and signal that `window_size` should be the entire pageable set (e.g. from+size). But I am against making that mandatory.

Reply:
> signal that window_size should be the entire pageable set 

I agree with that as well. So essentially, we document that `window_size` should be at least `from + size` and that pagination happens only within  `window_size` results, but we could for example return no results if `window_size` is not configured properly for pagination (e.g. `window_size: 10`, `size:5`, `from: 100`)

Reply:
@benwtrent So for re-rankers are you saying that `window_size` should be independent of from in this case? IE pagination is `from + size` always and `window_size` needs to accommodate this by being >=. I think I like this approach because it if I'm understanding correctly, pagination within a specific `window_size` will lead to consistent results. And if `window_size` changes we can document that pagination does not necessarily remain the same. The only concern I would have is users may have very large `window_size` to support this when they may not need them for the initial pages.